### DISCUSSION TITLE
On the efficiency of air in mobile platform.

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#1220] [Q&A]
I usually use air to develop IOS applications. One problem has always been very distressed. I found that when dealing with bitmap movement or bitmap sequence playback (2048 * 1536), I feel that the efficiency is much worse than Objective-C, especially on PNG pictures.

I don't know if the methods I use consume more resources. I don't know if you have encountered the same problems when doing these applications, or if there are better ways to improve the running efficiency of the program?

**Comment by ajwfrost** [comment]
What are you actually doing with the PNGs? Moving a bitmap around the screen would involve different actions depending on what rendering mode it's using: the best approach is probably to use Stage3D with this as a texture that you just move around, I'm not sure whether there is GPU acceleration in play otherwise.

If you can create a test case for it to demonstrate the kinds of activities you're doing, we can help to suggest the optimal way of approaching it?

thanks

**Comment by Jason0247** [comment]
@ajwfrost For example, I have many bitmap images that I want to read and display on the stage through the program.

When I see that the OC program is implemented in the same way, it seems that it can reach 25 frames per second on very old devices (such as iPad air generation 1), but I use air. The device must use the latest iPad (iPad air generation 1 will have a special card, even only 1 frame per second), and the speed is only 10 frames per second.

I don't know if air has a more efficient way to deal with the same problem, so I'm confused. I don't know if it's a procedural mechanism problem. Is there a way to overcome it.

thank you!

Demo link：http://keketuohai.rico.wang:8888/down/d3kBkNCIqefa

**Comment by Jason0247** [comment]
![image](https://user-images.githubusercontent.com/91530077/135594692-49cf6e03-e371-454f-b4c7-91195771f4c3.png)
